occjava/src/GC.i

```diff
@@ -16,11 +16,34 @@
  *
  */
  
- %{#include <GC_MakeArcOfCircle.hxx>%}
+%{
+#include <GC_MakeArcOfCircle.hxx>
+#include <GC_MakeSegment.hxx>
+#include <GCE2d_Root.hxx>
+#include <GCE2d_MakeSegment.hxx>
+%}
  
- class GC_MakeArcOfCircle {
-   public:
-   GC_MakeArcOfCircle(const gp_Pnt& P1,const gp_Pnt& P2,const gp_Pnt& P3);
-   GC_MakeArcOfCircle(const gp_Circ& Circ,const gp_Pnt& P1,const gp_Pnt& P2,const Standard_Boolean Sense);
-   const Handle_Geom_TrimmedCurve& Value() const;
- };
\ No newline at end of file
+class GC_MakeArcOfCircle {
+	public:
+	GC_MakeArcOfCircle(const gp_Pnt& P1,const gp_Pnt& P2,const gp_Pnt& P3);
+	GC_MakeArcOfCircle(const gp_Circ& Circ,const gp_Pnt& P1,const gp_Pnt& P2,const Standard_Boolean Sense);
+	const Handle_Geom_TrimmedCurve& Value() const;
+ };
+ 
+class GC_MakeSegment {
+    public:
+    GC_MakeSegment(const gp_Pnt& P1, const gp_Pnt& P2);
+ 	const Handle_Geom_TrimmedCurve& Value() const;
+ };
+ 
+class  GCE2d_Root {
+ 	 public:
```

Comment:
It's not virtual pure in OCE/OCC but IMHO it should be. So here I would add a virtual pure constructor, so no java code is generated to allow creation of this object.